src/Mollie/WC/Gateway/Abstract.php

```diff
@@ -1296,25 +1291,62 @@ public function getReturnRedirectUrlForOrder( WC_Order $order ) {
 
 			}
 
-			$payment = Mollie_WC_Plugin::getPaymentObject()->getActiveMolliePayment($order_id, false );
-
-			if ( ! $payment->isOpen() && ! $payment->isPending() && ! $payment->isPaid() && ! $payment->isAuthorized() ) {
-				Mollie_WC_Plugin::addNotice( __( 'Your payment was not successful. Please complete your order with a different payment method.', 'mollie-payments-for-woocommerce' ) );
-				// Return to order payment page
-				if ( method_exists( $order, 'get_checkout_payment_url' ) ) {
-					return $order->get_checkout_payment_url( false );
-				}
-			}
-
+            try {
+                $payment = $this->activePaymentObject($order_id, false);
+                if ( ! $payment->isOpen() && ! $payment->isPending() && ! $payment->isPaid() && ! $payment->isAuthorized() ) {
+                    notice(__('Your payment was not successful. Please complete your order with a different payment method.', 'mollie-payments-for-woocommerce'));
+                    // Return to order payment page
+                    if ( method_exists( $order, 'get_checkout_payment_url' ) ) {
+                        return $order->get_checkout_payment_url( false );
+                    }
+                }
+                do_action( Mollie_WC_Plugin::PLUGIN_ID . '_customer_return_payment_success', $order );
+            } catch (UnexpectedValueException $exc) {
+                notice( __('There was a problem when processing your payment. Please try again.', 'mollie-payments-for-woocommerce' ));
+                $exceptionMessage = $exc->getMessage();
+                $debugLine = __METHOD__ . " Problem processing the payment. {$exceptionMessage}";
+                debug($debugLine);
+                do_action( Mollie_WC_Plugin::PLUGIN_ID . '_customer_return_payment_failed', $order );
+            }
 		}
 
-		do_action( Mollie_WC_Plugin::PLUGIN_ID . '_customer_return_payment_success', $order );
 		/*
 		 * Return to order received page
 		 */
-
 		return $this->get_return_url( $order );
 	}
+    /**
+     * Retrieve the payment object
+     *
+     * @return Mollie_WC_Payment_Object
+     */
+    protected function paymentObject()
+    {
+        return Mollie_WC_Plugin::getPaymentObject();
+    }
+
+    /**
+     * Retrieve the active payment object
+     *
+     * @param $orderId
+     * @param $useCache
+     * @return Payment
+     * @throws UnexpectedValueException
+     */
+    protected function activePaymentObject($orderId, $useCache)
+    {
+        $paymentObject = $this->paymentObject();
+
+        $activePaymentObject = $paymentObject->getActiveMolliePayment($orderId, $useCache);
```

Comment:
Just for code style, when you have multiple assignments which are related somehow it common to not have any blank line between them. This help in understanding the code because you group things which belongs together.